src/urllib3/exceptions.py

```diff
@@ -160,10 +161,14 @@ class ConnectTimeoutError(TimeoutError):
     pass
 
 
-class NewConnectionError(ConnectTimeoutError, PoolError):
+class NewConnectionError(ConnectTimeoutError, HTTPError):
     """Raised when we fail to establish a new connection. Usually ECONNREFUSED."""
 
-    pass
+    pool: "HTTPConnection"
```

Comment:
Can you please update `pool` to `conn` here and below?

Reply:
I will do this and I am going to add a deprecated property as @sethmlarson suggested here https://github.com/urllib3/urllib3/pull/2271#discussion_r653544455.

Shouldn't we go for deprecation?

Reply:
As I mentioned above I don't think a deprecation makes sense unless we keep the `PoolError` inheritance. It's possible, but not worth it in my opinion. I'll let Seth decide

Reply:
My thought was to have a deprecation only on the accessor property `pool` and to remove pool completely from the constructor as it's a much bigger burden.

The likelyhood of someone accessing the actual object attached to the error is higher than someone using a keyword argument to construct an instance of this error (likely we're mostly the only ones creating these errors).

Maybe we can have a more aggressive deprecation period for these types of changes too, maybe say "in a later 2.x release" instead of "3.0.0"?